### DISCUSSION TITLE
how to send button and axis data to host for a joystick?

### DISCUSSION THREAD

**Discussion by skreamz** [hathach/tinyusb#602] [Q&A] [answered]
Hi all, this is my first USB project and I've got a descriptor set up and my computer is seeing my device as a USB joystick. the only issue I have is, I don't know how to send the data (button presses and axis info) to the host, I've searched, but all I can find is mouse and keyboard examples. so I've no idea what function or how to structure the data needed. any help would be grateful.

Thank you.

**Comment by chegewara** [comment]
All depends how is descriptor prepared. Maybe here you can find inspiration:
https://eleccelerator.com/tutorial-about-usb-hid-report-descriptors/

**Comment by skreamz** [comment]
thank you, I actually started down that road by guesswork.

i guess I found the issue, 

tud_hid_ready() always returns false.

any idea what could cause this?

Thanks again.

**Comment by hathach** [comment, marked as answer]
have you try one of the stock hid example such as hid_composite (with/wihtout freertos), try to see if that works for you and start there. Also, when asking for advice, you should provide more about your set up (OS, mcu, board, toolchain IDE etc ...). That will save time for people from doing guessing and generic answer.

**Reply by skreamz** [reply]
my apologies, it's my first time working with microcontrollers. was using the pi pico. got it to work, recreated the project, but reused all the same code, and all of a sudden it's working, anyways thank you all for your time.
